xdsl/xdsl_opt_main.py

```diff
@@ -352,3 +336,6 @@ def print_to_output_stream(self, contents: str):
         else:
             output_stream = open(self.args.output_file, 'w')
             output_stream.write(contents)
+
+    def get_input_name(self):
+        return self.args.input_file or 'stdin'
```

Comment:
This also seems to be an independent change.

Reply:
I added this to factor out some common operation. We need to pass the name of the input to the parser now, so I thought this might be a good place to put it?